### PR TITLE
Fetch KeepAlive Quota Tests

### PR DESCRIPTION
**Pull request by Brandr0id**
Initial add of quota tests for keepalive fetch requests.

### CODE REVIEW

**Review by igrigorik — APPROVED** [review]
LGTM!

File: fetch/api/request/request-keepalive-quota.html [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            // prevent any of the Fetch's from finishing in this window.
+            var trickleURL = "../resources/trickle.py?count=1&ms=";
+            var standardDelay = 1000;
+
+            // We should expect 64KB of rolling quota for any type of keep-alive request sent.
+            var expectedQuota = 65536;
+
+            function CreateKeepAliveRequest(delay, bodySize) {
+                // Create a body of the specified size that's filled with *'s
+                var requestBody = Array(bodySize).fill('*').join("");
```
Comment (annevk) [inline fetch/api/request/request-keepalive-quota.html]: Maybe use `"*".repeat(bodySize)` instead?

File: fetch/api/request/request-keepalive-quota.html [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        <script>
+            "use strict";
+
+            // We want to ensure that our keepalive requests hang slightly before completing so we can validate
+            // the effects of a rolling quota. To do this we will utilize trickle.py with a 1s delay. This should
+            // prevent any of the Fetch's from finishing in this window.
+            var trickleURL = "../resources/trickle.py?count=1&ms=";
+            var standardDelay = 1000;
+
+            // We should expect 64KB of rolling quota for any type of keep-alive request sent.
```
Comment (annevk) [inline fetch/api/request/request-keepalive-quota.html]: KiB

**Review by annevk — COMMENTED** [review]
These look good, just two minor nits. Curious, did you test them against an implementation?

File: fetch/api/request/request-keepalive-quota.html (hunk 2 above)
Comment (Brandr0id) [inline fetch/api/request/request-keepalive-quota.html]: Yeah, validated using Edge's implementation. Rolling keepalive quotas will be available for fetch w/keepalive and sendBeacon in an upcoming insider build.

**Comment by Brandr0id** [comment]
Yeah, validated using Edge's implementation. Rolling keepalive quotas will be available for fetch w/keepalive and sendBeacon in an upcoming insider build.

**Comment by annevk** [comment]
Either something is wrong with the build system or these tests are flaky somehow. Not sure what is going on. I asked @jgraham to look into it.

**Comment by annevk** [comment]
Okay, I think the problem is that you're putting your tests in w3c:MicrosoftEdge rather than master. That is what causes Travis to apply hundreds of commits and essentially get lost.

**Comment by jgraham** [comment]
This is breaking on travis because you are trying to merge into the MicrosoftEdge branch rather than master. We can probably fix that in the bot, but why are you trying to do that?

**Comment by thejohnjansen** [comment]
We discussed this at the F2F last week, but I realize we did not come to a conclusion. Our plan was that we would use the MicrosoftEdge branch as the initial PR location where we'd do a Microsoft code review in the public eye. Then we'd do a 2nd PR from there merging it into master. We are trying to save other reviewers time so they don't review a change from Microsoft unnecessarily. I may not have been clear when we were discussing this and maybe I misunderstood suggestions. If there is a better way to save everyone time and get the public review from MS done, I'm happy to implement it.

**Comment by annevk** [comment]
I don't mind reviewing tests from Microsoft directly. I also review them from Google and Apple, and sometimes even Mozilla.

But I also don't mind if you want that kind of setup, but then the tooling should probably be updated to take that into account and not start pinging folks for pull requests that are not against master.

**Comment by thejohnjansen** [comment]
Thanks Anne. I think for this PR, we will change it to just be against Master since the tools are not updated yet. Moving forward, we'll evaluate whether we should do some fork, modify the tools, or simply PR into Master and worry about code reviews there.

**Comment by Brandr0id** [comment]
I submitted https://github.com/w3c/web-platform-tests/pull/4878 against master for this change. @igrigorik @annevk

**Comment by Brandr0id** [comment]
Closing PR in favor of new one against master.

**Pull request closed without merging**